### PR TITLE
Set up invariant test

### PR DESCRIPTION
**Pull request by offerijns**

### CODE REVIEW

File: test/accounts/PoolManager.sol [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  bytes16[] public allTranches;
+  mapping(bytes16 => uint64) public trancheIdToPoolId;
+
+  constructor(MockHomeConnector connector_) public {
+    connector = connector_;
+  }
+
+  function addPool(uint64 poolId) public {
+    connector.addPool(poolId);
+
```
Comment (NunoAlexandre) [inline test/accounts/PoolManager.sol:22]:
```suggestion
```

File: test/accounts/PoolManager.sol [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  function addPool(uint64 poolId) public {
+    connector.addPool(poolId);
+    
+    allPools.push(poolId);
+  }
+
+  // uint8 so the fuzzer more often finds valid array indices
+  function addTranche(uint8 poolIndex, bytes16 trancheId, string memory tokenName, string memory tokenSymbol, uint128 price) public {
+    uint64 poolId = allPools[poolIndex];
+    vm.assume(poolId > 0);
```
Comment (NunoAlexandre) [inline test/accounts/PoolManager.sol:29]: Why do we need this `assume`?

**Review by NunoAlexandre — DISMISSED** [review]
This is nice 👌

File: test/accounts/PoolManager.sol (hunk 2 above)
Comment (offerijns) [reply test/accounts/PoolManager.sol:29]: The fuzzer will generate random uint8 `poolIndex` values, and there's no guarantee that these actually exist. This basically checks that.

**Review by NunoAlexandre — APPROVED** [review]

**Pull request merged**